### ISSUE TITLE
Files are not being uploaded 

### ISSUE THREAD

**Issue by Bebra777228** [gradio-app/gradio#9646]
### Describe the bug

In version 5 of Gradio, some files do not load.

### Have you searched existing issues?  🔎

- [x] I have searched and found no existing issues

### Reproduction

```python
zip_file = gr.File(label="Zip-File", file_types=[".zip"], file_count="single")
```

### Screenshot

![image](https://github.com/user-attachments/assets/64611d3d-32f3-4217-88b1-3757755fd59d)
![image](https://github.com/user-attachments/assets/24ee4121-6264-458c-924f-a99a716d4f26)

### Logs

```shell
Traceback (most recent call last):
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\queueing.py", line 622, in process_events
    response = await route_utils.call_process_api(
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\route_utils.py", line 323, in call_process_api
    output = await app.get_blocks().process_api(
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\blocks.py", line 2012, in process_api
    inputs = await self.preprocess_data(
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\blocks.py", line 1711, in preprocess_data
    processed_input.append(block.preprocess(inputs_cached))
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\components\file.py", line 163, in preprocess
    return self._process_single_file(payload)
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\components\file.py", line 132, in _process_single_file
    raise Error(
gradio.exceptions.Error: "Invalid file type. Please upload a file that is one of these formats: ['.zip']"
Traceback (most recent call last):
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\queueing.py", line 622, in process_events
    response = await route_utils.call_process_api(
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\route_utils.py", line 323, in call_process_api
    output = await app.get_blocks().process_api(
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\blocks.py", line 2012, in process_api
    inputs = await self.preprocess_data(
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\blocks.py", line 1711, in preprocess_data
    processed_input.append(block.preprocess(inputs_cached))
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\components\file.py", line 163, in preprocess
    return self._process_single_file(payload)
  File "C:\Users\Politrees\Desktop\PolGen\env\lib\site-packages\gradio\components\file.py", line 132, in _process_single_file
    raise Error(
gradio.exceptions.Error: "Invalid file type. Please upload a file that is one of these formats: ['.pth']"
```

### System Info

```shell
Operating System: Windows
gradio version: 5.0.1
gradio_client version: 1.4.0
```

### Severity

Blocking usage of gradio

**Comment by dawoodkhan82** [comment]
It seems to be working fine for me. Maybe this is a Windows issue? @yvrjsharma Do you mind testing?

**Comment by yvrjsharma** [comment]
I am unable to reproduce this issue on my Windows 11 machine (Python 3.12.3, gradio 5.0.1). Reproducible sample provided above is working for me @dawoodkhan82

**Comment by Bebra777228** [comment]
I have Windows 10 and Python 3.10.
I don't know what else to explain here, this file upload works fine on Gradio 4.44.1, but it doesn't work on version 5.0.1.

**Comment by coca-create** [comment]
My application cannot upload srt files,too.
I use python3.11.8 , so I downgrade version 4.44.1. well, but I want to use 5.0.1 because of the new GUI of 5.0.1.

**Comment by jasongzy** [comment]
This issue originates from the enforced `file_type` check introduced in 5.0.
However, the implementation has problems when dealing with unconventional file types.
In fact, `mimetypes.guess_type()` is used internally, but it cannot recognize file extensions other than conventional ones:

```python
import mimetypes
print(mimetypes.guess_type("foo.jpg"))
# ('image/jpeg', None)
print(mimetypes.guess_type("foo.pth"))
# (None, None)
```

Here are the implementations of gardio:

https://github.com/gradio-app/gradio/blob/3972a1b30f15f95a2977ba5fd0a6515e338266b3/gradio/components/file.py#L129-L134

https://github.com/gradio-app/gradio/blob/3972a1b30f15f95a2977ba5fd0a6515e338266b3/client/python/gradio_client/utils.py#L690-L693

https://github.com/gradio-app/gradio/blob/3972a1b30f15f95a2977ba5fd0a6515e338266b3/client/python/gradio_client/utils.py#L668-L671

So basically, it will always fail to pass the `file_type` check (even when a correct file is uploaded) and raise an error when we use something like:

```python
gr.File(file_types=[".pth"])
```

**Comment by freddyaboulton** [comment]
Thanks for the detailed write-up @jasongzy . Would you like to open a PR? In the case the user specifies a file extension in `allowed_files` we don't need to guess the mime type of the allowed file. We can just check if the file extensions match.

**Comment by meefyGH** [comment]
Windows 10 
Python 3.10.
gradio==5.0.1 

file_upload = gr.File(label="上传文件", file_types=['.pdf', '.txt', '.doc'])

this file upload works fine on Gradio 4.43.0, but it doesn't work on version 5.0.1.

**Comment by all4you** [comment]
> Thanks for the detailed write-up @jasongzy . Would you like to open a PR? In the case the user specifies a file extension in `allowed_files` we don't need to guess the mime type of the allowed file. We can just check if the file extensions match.

I have edited the is_valid_file function to this,and then it works fine:
def is_valid_file(file_path: str, file_types: list[str]) -> bool:
    file_extension = os.path.splitext(file_path)[1]
    return file_extension in file_types
